youtubeExtension/options.js

```diff
@@ -0,0 +1,86 @@
+function saveOptions() {
+
+  const checkboxIds = ['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'];
+  let options = {};
+
+  // loop through each checkbox id and save its state
+  checkboxIds.forEach(id => {
+    options[id] = document.getElementById(id).checked;
+  });
+
+  chrome.storage.sync.set(options);
+
+}
+
+function saveRange(){
+  const range = document.getElementById("my-range");
+  chrome.storage.sync.set({'my-range': slider.value});
+  console.log({'my-range': slider.value});
+}
+
+function restoreOptions() {
+
+  const checkboxIds = ['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'];
+  // retrieve the saved state of each checkbox and set it
+  chrome.storage.sync.get(checkboxIds, (items) => {
+    checkboxIds.forEach(id => {
+      document.getElementById(id).checked = items[id] || false;
+    });
+  });
+
+}
+
+function restoreRange(){
+  chrome.storage.sync.get('my-range', (items) => {
+    const slider = document.getElementById("my-range");
+    const output = document.getElementById("demo");
+
+    if (items['my-range'] !== undefined && slider && output) {
+      slider.value = items['my-range'];
+      output.textContent = items['my-range'];
+    }
+  });
+}
+
+// add event listener to save options when the DOM is fully loaded
+document.addEventListener('DOMContentLoaded', () => {
+
+  restoreOptions();
+  restoreRange();
+
+  const checkboxIds = ['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'];
+});
+
+let ReviewThemes =[];
+
+// event listener to log when save button is clicked
+document.getElementById('saveButton').addEventListener('click', () => {
+  console.log("Save button clicked");
+  saveOptions();
+  saveRange();
+  // log option settings
+  chrome.storage.sync.get(['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'], function(items) {
+    // `items` is an object containing all synced items (if `null` is passed as the first argument)
+    console.log("All items in chrome.storage.sync:", items);
+
+    const ReviewThemes = items;
+    console.log(ReviewThemes);
+
+  });
+
+});
+
+
+
+const slider = document.getElementById("my-range");
```

Comment:
I assume this works, so maybe it's not a problem, but I'm generally nervous having code like this at the top level of a js file like this. This code will run immediately. But it assumes there's an element called `my-range`, and it's possible that doesn't exist yet (if this js runs before the DOM has loaded).

Normally, you'd put this code inside a function that runs after the document has finished loading, so you're sure things won't run in the wrong order.